src/bootloader/bootloaderlite.h

```diff
@@ -0,0 +1,41 @@
+#ifndef BOOTLOADERLITE_H_
+#define BOOTLOADERLITE_H_
+
+#include "bootloader/bootloader.h"
+#include "libaktualizr/config.h"
+#include "rollbacks/rollback.h"
+
+class INvStorage;
+
+class BootloaderLite : public Bootloader {
+ public:
+  BootloaderLite(BootloaderConfig config, INvStorage& storage);
+  virtual ~BootloaderLite() {}
+  void setBootOK() const;
+  void updateNotify() const;
+
+  // Reboot handling (uses storage)
+  //
+  // Note: will only flag a reboot if it was flagged for detection with
+  // `rebootFlagSet()`.
+  // Also, `rebootDetected()` will continue to return true until the flag
+  // has been cleared, so that users can make sure that appropriate actions
+  // in reaction to the reboot have been processed.
+  bool supportRebootDetection() const;
+  bool rebootDetected() const;
+  void rebootFlagSet();
+  void rebootFlagClear();
+  void reboot();
+  void rebootFake();
+
+ private:
+  const BootloaderConfig config_;
+
+  INvStorage& storage_;
+  std::unique_ptr<Rollback> rollback_;
+  boost::filesystem::path reboot_sentinel_;
+  std::string reboot_command_;
+  bool reboot_detect_supported_{false};
```

Comment:
why do we need to duplicate the fields from the class we inherit from?

Reply:
because my understanding was that we would not reuse anything from the base class.

If that is not the case, then really we should have an abstract interface and clients providing their bootloader specific implementations. why would you want to depend on the libaktualizr more than what is necesseray (ie, we only need the interfaces)

Reply:
>why would you want to depend on the libaktualizr implementation more than what is necessary (ie, we only need the interfaces)

I don't want, but we already depend on ostreemanager that wants an object that implements Bootloader. If you prefer you can skip inheritance then.